packages/python/accern_data/accern_data.py

```diff
@@ -245,34 +265,66 @@ def max_date(
         # Second last date.
         return max(dates[:-1]).strftime(DT_FORMAT)
 
-    def do_init(self, is_first_day: bool) -> None:
+    def do_init(self, indicator: ProgressIndicator) -> None:
         fname = self.get_path(self._is_by_day)
-        if is_first_day or self._is_by_day:
+        if self._begining_of_file or self._is_by_day:
             pd.DataFrame([], columns=self._cols).to_csv(
```

Comment:
why are you creating an empty file. just wait until the first real write happens and check if cols is None in which case you have to write the header

Reply:
(and set cols from the df you have there)